tappio/writer.py

```diff
@@ -45,7 +48,7 @@ def write(self, *tokens):
         for token in tokens:
             if self.should_put_space_between(self.prev_token, token):
                 self.stream.write(" ")
-            self.stream.write(str(token))
+            self.stream.write(unicode(token))
```

Comment:
I'm pretty sure there's an encoding bug in the making here somewhere. I need to test this with Finnish (äöå) input.

Reply:
Since I'm pretty much ignorant of the whole tokenisation thing I'll leave this to you. If the purpose of the `str` call is to invoke token's `__str__` method then I'm unsure as to whether the return value was intended to be bytes or text (though it should surely _always_ be text, shouldn't it?). Perhaps an explicit method like `to_bytes` and/or `to_str`?

Reply:
This one probably is your call. Thanks again for the book-fodder! Happy to try and swing you a freebie Nutshell from O'Reilly if you would like?